src/router.py

```diff
@@ -0,0 +1,159 @@
+    # Checks if the packet is for me
+    if dst != leg.mac and dst != BROADCAST:
+        return None
+
+    # Parse arp layer
+    if eth.ether_type == pascy.EthernetLayer.ARP_ETHER_TYPE:
+        arp = pascy.l2.ArpLayer()
+        arp.deserialize(pkt_raw)
+        pkt_raw = pkt_raw[len(arp):]
+        eth.connect_layer(arp)
+
+    # Parse ip layer
+    elif eth.ether_type == pascy.EthernetLayer.IPV4_ETHER_TYPE:
+        ip = pascy.l3.IpLayer()
+        ip.deserialize(pkt_raw)
+        pkt_raw = pkt_raw[len(ip):]
+        
+        # Parse icmp layer
+        if ip.protocol == pascy.IpLayer.ICMP_PROTOCOL_NUMBER:
+            icmp = pascy.l3.IcmpLayer()
+            icmp.deserialize(pkt_raw)
+            pkt_raw = pkt_raw[len(icmp):]
+            ip.connect_layer(icmp)
+
+        eth.connect_layer(ip)
+
+    if len(pkt_raw) > 0:
+        raw = pascy.RawLayer()
+        raw.load = pkt_raw
+        eth.connect_layer(raw)
+
+    return eth
+
+def response_arp(pkt, leg : RouterLeg):
+    resp_pkt = pascy.l2.EthernetLayer() / pascy.l2.ArpLayer() / pascy.RawLayer()
+    resp_pkt["Ethernet"].dst = pkt["Ethernet"].src
+    resp_pkt["Ethernet"].src = leg.mac
+
+    resp_pkt["ARP"].copy_layer(pkt["ARP"])
+    resp_pkt["ARP"].operation = pascy.ArpLayer.OP_IS_AT
+    resp_pkt["ARP"].sender_hardware_addr = leg.mac
+    resp_pkt["ARP"].sender_protocol_addr = leg.ip
+    resp_pkt["ARP"].target_hardware_addr = pkt["Ethernet"].src
+    resp_pkt["ARP"].target_protocol_addr = pkt["ARP"].sender_protocol_addr
+
+    leg.raw_socket.sendall(resp_pkt.build())
+    print("Sent ARP reply.")
+
+def response_icmp(pkt, leg : RouterLeg):
+    resp_pkt = pascy.EthernetLayer() / (pascy.IpLayer() / pascy.IcmpLayer())
+    resp_pkt.connect_layer(pascy.RawLayer())
+    resp_pkt["Ethernet"].dst = pkt["Ethernet"].src
+    resp_pkt["Ethernet"].src = leg.mac
+
+    resp_pkt["IP"].copy_layer(pkt["IP"])
+    resp_pkt["IP"].dst = pkt["IP"].src
+    resp_pkt["IP"].src = leg.ip
+    resp_pkt["IP"].calc_checksum()
+
+    resp_pkt["ICMP"].type = pascy.IcmpLayer.ECHO_REPLY
+    resp_pkt["ICMP"].rest = pkt["ICMP"].rest
+    resp_pkt["ICMP"].calc_checksum()
+
+    resp_pkt["Raw"].load = pkt["Raw"].load
+
+    leg.raw_socket.sendall(resp_pkt.build())
+    print("Sent ICMP reply.")
+
+def forward_packet(pkt, leg : RouterLeg):
+    dst_ip = socket.inet_ntoa(pkt["IP"].dst)
+    pkt["Ethernet"].src = leg.mac
+    pkt["Ethernet"].dst = ARP_TABLE[dst_ip]
+    pkt["IP"].ttl -= 1
+
+    leg.raw_socket.sendall(pkt.build())
+    print("Message forwarded")
+
+@lru_cache()
+def get_leg_by_socket(s) -> RouterLeg:
+    for leg in ROUTER_LEGS:
+        if leg.raw_socket == s:
+            return leg
+
+    return None
+
+@lru_cache()
+def get_leg_by_ip(ip) -> RouterLeg:
+    for leg in ROUTER_LEGS:
+        if leg.subnet in ip:
+            return leg
+
+    return None
+
+def main():
+    ROUTER_LEGS.append(RouterLeg('net1', '1.1.1', '02:42:05:66:DC:60', '1.1.1.1'))
```

Comment:
What is `1.1.1`?
Did you mean `1.1.1.0/24`?
You should explicitly state the subnet mask.
Many routing options differ in subnets